### PR TITLE
Alerting: Add fgac support to alert list panel

### PR DESCRIPTION
**Pull request by konrad147**
**What this PR does / why we need it**:
Add basic permission checking to the alert list panel and alerting query runner

### CODE REVIEW

File: public/app/plugins/panel/alertlist/UnifiedAlertList.tsx [hunk 1]
```diff
@@ -60,6 +62,13 @@ export function UnifiedAlertList(props: PanelProps<UnifiedAlertListOptions>) {
 
   const noAlertsMessage = rules.length ? '' : 'No alerts';
 
+  if (
+    !contextSrv.hasPermission(AccessControlAction.AlertingRuleRead) &&
+    !contextSrv.hasPermission(AccessControlAction.AlertingRuleExternalRead)
+  ) {
+    return <Alert title="Permission required">Sorry, you don not have required permissions to read alert rules</Alert>;
```
Comment (gillesdemey) [inline public/app/plugins/panel/alertlist/UnifiedAlertList.tsx:69]: typo :) 

```suggestion
    return <Alert title="Permission required">Sorry, you do not have the required permissions to read alert rules</Alert>;
```

File: public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts [hunk 2]
```diff
@@ -29,6 +31,10 @@ export class UnifiedAlertStatesWorker implements DashboardQueryRunnerWorker {
       return false;
     }
 
+    if (!contextSrv.hasPermission(AccessControlAction.AlertingRuleRead)) {
```
Comment (gamab) [inline public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts:34]: No need to check `AccessControlAction.AletingRuleExternalRead`?

**Review by gamab — APPROVED** [review]
I'm a bit out of my comfort zone here, as I don't understand where/why `canWork()` is used, but rbac wise the permission check is correct.

File: public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts (hunk 2 above)
Comment (IevaVasiljeva) [reply public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts:34]: A further question here - what's the granularity for `AletingRuleExternalRead` action? Can it be scoped per dashboard? If so, you'll probably want to check whether the user has been granted `AletingRuleExternalRead` action for this particular dashboard (we've done similar things with [teams](https://github.com/grafana/grafana/blob/main/public/app/features/teams/TeamList.tsx#L76-L78), data sources etc).

**Review by IevaVasiljeva — COMMENTED** [review]
Same question as what @gamab asked, plus one more question in line.

Another thing to keep in mind is that `contextSrv.hasPermission` returns `true` if FGAC is disabled. If I remember correctly, alerting is accessible for viewers when FGAC is disabled, so it's not a problem. Just wanted to double-check :)

File: public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts (hunk 2 above)
Comment (konrad147) [reply public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts]: I added check for `AccessControlAction.AletingRuleExternalRead`
`AccessControlAction.AletingRuleExternalRead` has a scope of `datasource:*` so if the user doesn't have access to a data source then they won't see alerts from that data source
The `UnifiedAlertStateWorker` is only used to show the alerting status indicator for each panel with alerts assigned. If the `canWork` method returns false then the indicator is not displayed

**Comment by konrad147** [comment]
> Same question as what @gamab asked, plus one more question in line.
> 
> Another thing to keep in mind is that `contextSrv.hasPermission` returns `true` if FGAC is disabled. If I remember correctly, alerting is accessible for viewers when FGAC is disabled, so it's not a problem. Just wanted to double-check :)

Yes, it should be accessible for Viewers when RBAC is disabled :)

**Review by gillesdemey — APPROVED** [review]
New changes look 👌

**Review by peterholmberg — APPROVED** [review]
Agreed, looks good!

**Pull request merged**